SherpaFrontEnd.Tests/Pages/TemplatesTest.cs

```diff
@@ -9,29 +12,49 @@ namespace BlazorApp.Tests.Pages;
 public class TemplatesTest
 {
     private readonly Mock<ITemplateService> _mockService;
+    private readonly TestContext _testContext;
+    private readonly FakeNavigationManager _navMan;
+    private TemplateWithoutQuestions[] _templates;
+    const string TemplatesPage = "templates";
 
     public TemplatesTest()
     {
+        _testContext = new TestContext();
         _mockService = new Mock<ITemplateService>();
+        _testContext.Services.AddBlazoredModal();
+        _testContext.Services.AddSingleton<ITemplateService>(_mockService.Object);
+        _navMan = _testContext.Services.GetRequiredService<FakeNavigationManager>();
     }
 
     [Fact]
-    public void Component_should_print_the_result_of_calling_getAllTemplates_of_service()
+    public void ComponentShouldPrintTheResultOfCallingGetAllTemplatesOfService()
     {
-        var ctx = new TestContext();
-        var templates = new[] { new TemplateWithNameAndTime("Hackman Model", 30) };
-        _mockService.Setup(service => service.GetAllTemplates()).ReturnsAsync(templates);
-        ctx.Services.AddSingleton<ITemplateService>(_mockService.Object);
-        var component = ctx.RenderComponent<Templates>();
-        
-        var titleElement = component.FindAll("h2").FirstOrDefault(element => element.InnerHtml.Contains("Hackman Model"));
+        _templates = new[] { new TemplateWithoutQuestions("Hackman Model", 30) };
+        _mockService.Setup(service => service.GetAllTemplates()).ReturnsAsync(_templates);
+        var component = _testContext.RenderComponent<Templates>();
+
+        var titleElement = component.FindAll("h2")
+            .FirstOrDefault(element => element.InnerHtml.Contains("Hackman Model"));
         var timeElement = component.FindAll("p").FirstOrDefault(element => element.InnerHtml.Contains("30 min"));
-        var threeDotsButton = component.FindAll("button")
-            .FirstOrDefault(element => element.InnerHtml.Contains("Open contextual menu for Hackman Model"));;
 
-        
+
         Assert.NotNull(titleElement);
         Assert.NotNull(timeElement);
-        Assert.NotNull(threeDotsButton);
+    }
+
+    [Fact]
+    public async Task ShouldBeAbleToClickOnATeamAndNavigateToItsOwnPage()
```

Comment:
Question here. This test is saying that it should be able to click on a team and navigate to its own page. However aren't we clicking on template rather than team?

Reply:
Right, typo